### ISSUE TITLE
Input size for F1 Score

### ISSUE THREAD

**Issue by KacperKubara** [tensorflow/addons#1953]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow version and how it was installed (source or binary): binary
- TensorFlow-Addons version and how it was installed (source or binary): binary
- Python version: 3.7
- Is GPU used? (yes/no): no

**Describe the bug**
Hi, I wanted to ask what input shape is allowed for F1 Score metric in the library. Should the y_true and y_pred vectors be one hot encoded, e.g. [batch_size, no_labels], or contain label of the prediciton, e.g. [batch_size]? Can I make it work with raw softmax values as well? Can't find any reference to this in docs.

**Comment by WindQAQ** [comment]
cc @AakashKumarNain who might understand metrics more.

**Comment by AakashKumarNain** [comment]
Thanks @WindQAQ 

@KacperKubara It should work in both the cases, i.e. OHE as well as with the `softmax` scores. If something isn't working, please let us know. We will also update the documentation for F1 score in the next version.  Thank you for the inputs

**Comment by KacperKubara** [comment]
That's good to know, thanks @AakashKumarNain. I've had some issues with F1 score but it seems that it was because of my misunderstanding. If I will encounter any bugs, I will open another issue.

I am happy to hear that docs will updated for F1 scores. On the sidenote, while the TFA seems to be a great addition to TF, the lack of docs (especially for metrics) refrained me from actually using it and looking for alternatives. The implementation code for metrics is also quite complicated and uses a lot of inheritance which makes it quite difficult to figure out what's going on by yourself. I will close this issue for now.

**Comment by AakashKumarNain** [comment]
@KacperKubara thanks for the feedback. We will take this into account and will try our best to improve the end-user experience. On a side not, if you think that the documentation of particular functionality in TFA isn't proper, please feel free to open an issue and we will look into it.
